docs/tutorials/dynamics_backend.rst

```diff
@@ -0,0 +1,489 @@
+
+    rabi0_data.figure(0)
+
+.. jupyter-execute::
+
+    rabi1_data.figure(0)
+
+Observe the updated parameters for qubit 0.
+
+.. jupyter-execute::
+
+    pd.DataFrame(**cals.parameters_table(qubit_list=[0, ()], parameters="amp"))
+
+5.4 Rough Drag parameter calibration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Run rough Drag parameter calibration for the ``X`` and ``SX`` gates. This follows the same procedure
+as above.
+
+.. jupyter-execute::
+
+    from qiskit_experiments.library.calibration import RoughDragCal
+    
+    cal_drag0 = RoughDragCal(0, cals, backend=backend, betas=np.linspace(-20, 20, 15))
+    cal_drag1 = RoughDragCal(1, cals, backend=backend, betas=np.linspace(-20, 20, 15))
+    
+    cal_drag0.set_experiment_options(reps=[3, 5, 7])
+    cal_drag1.set_experiment_options(reps=[3, 5, 7])
+    
+    cal_drag0.circuits()[5].draw(output="mpl")
+
+.. jupyter-execute::
+
+    %%time
+    drag0_data = cal_drag0.run().block_for_results()
+    drag1_data = cal_drag1.run().block_for_results()
+
+.. jupyter-execute::
+
+    drag0_data.figure(0)
+
+
+.. jupyter-execute::
+
+    drag1_data.figure(0)
+
+The updated calibrations object:
+
+.. jupyter-execute::
+
+    pd.DataFrame(**cals.parameters_table(qubit_list=[0, ()], parameters="amp"))
+
+6. Simulating 2 qubit interaction characterization via the |CRHamitonian| experiment
+------------------------------------------------------------------------------------
+
+Finally, simulate the |CRHamitonian| characterization experiment.
+
+First, we further configure the backend to run this experiment. This requires: 
+
+- Adding the custom gate used in the experiment as a valid instruction in the
+  :class:`~qiskit.transpiler.Target`. This ensures that the validation checks on the submitted jobs
+  recognize the gate as a valid instruction.
+- Defining the control channel map, which is a dictionary mapping control-target qubit index pairs
+  (given as a tuple) to the control channel index used to drive the corresponding cross-resonance
+  interaction. This is required by the experiment to determine which channel to drive for each
+  control-target pair.
+
+.. jupyter-execute::
+
+    # add the gate to the target
+    from qiskit_experiments.library import CrossResonanceHamiltonian
+    backend.target.add_instruction(
+        instruction=CrossResonanceHamiltonian.CRPulseGate(width=Parameter("width")), 
+        properties={(0, 1): None, (1, 0): None}
+    )
+    
+    # set the control channel map
+    backend.set_options(control_channel_map={(0, 1): 0, (1, 0): 1})
+
+Build the characterization experiment object, and set the instruction map in the transpilation
+options to use the single qubit gates calibrated above.
+
+.. jupyter-execute::
+
+    cr_ham_experiment = CrossResonanceHamiltonian(
+        qubits=(0, 1), 
+        flat_top_widths=np.linspace(0, 5000, 17), 
+        backend=backend
+    )
+    
+    cr_ham_experiment.set_transpile_options(inst_map=cals.default_inst_map)
```

Comment:
Since the dynamics backend is a V2 wrapper, this will run into the [known bug](https://github.com/Qiskit/qiskit-terra/issues/9489). You can export the instmap from calibration and use [this method](https://github.com/Qiskit/qiskit-terra/blob/126c9ba1e3b1dc1486b0191de759bd756ff91a28/qiskit/transpiler/target.py#L416) to mutate backend target.

Reply:
What is the correct way of doing this? I've tried:
```
backend.target.update_from_instruction_schedule_map(cals.get_inst_map())
```

but I get the error:
```
KeyError: "Provided qarg: '(0,)' not in this Target for x"
```

Reply:
hmm `Target` is not well supported by `Calibrations` at this moment. Indeed your calibration instance have several gates that is not predefined in your target. Probably this trick works

https://github.com/Qiskit/qiskit-experiments/blob/main/qiskit_experiments/calibration_management/basis_gate_library.py#L211

namely you can instantiate the template library with the limited set of basis gates (`basis_gates=["sx"]` in your example) that is defined by your target instance.

Reply:
So if I add a dummy instruction, e.g. ``backend.target.add_instruction(ZGate())``, the `'coupling graph not connected'` error goes away, and the code runs, including the line:
```
backend.target.update_from_instruction_schedule_map(cals.get_inst_map())
```
I've done some cursory checks of `backend.target.instructions` and `backend.target.instruction_properties`, and it seems like the calibrations have been correctly updated.

So, that's all good, but I need to figure out how to get rid of the `CouplingError: 'coupling graph not connected'` error without the hack of adding some gate that applies to all qubits (like the `ZGate`). Any ideas?

Reply:
After discussing with @wshanks:
- There appears to be no way around this.
- As the lesser hack, I've also added `CXGate` as an instruction in the target configuration cell, along with a note explaining that this is being done so that that device registers as "connected" in validation checks.

Reply:
Well, I didn't know a way around without more investigation at least. My summary of what is happening is:

* qiskit-experiments always runs `transpile` when generating circuits.
* Some pass in the default pass manager wants to look at the full device and expects that the qubits in the device to all be part of one connected graph.
* With BackendV2/Target, the only way to specify the connectivity is by defining two qubit gates between pairs of qubits (you can't specify connectivity directly outside of gates).
* Defining a global gate (without specific qubit indices) causes this connectivity check to get bypassed (this lets just adding `ZGate` or `CXGate` without qubit indices work).

The solutions that I can think of besides adding a global gate are:

* Add `CXGate` (or other two qubit gate) with a specific qubit pair with a comment that this is how you define the connectivity of the qubits in a V2 backend. Dan tried this but ran into another transpiler error about reversing the direction of a gate.
* Update qiskit-experiments to allow more flexibilty with the transpilation (custom pass manager). With this, dynamics could avoid whatever transpiler pass is causing the problem currently. This has been discussed but is a longer term solution.
* Figure out which transpiler pass is the problem and see if it can be bypassed by setting a transpile option on the experiment.

@nkanazawa1989 Do you have any thoughts on those?

Also, one follow up thought -- maybe DynamicsBackend could do something to make this easier for the user? Maybe DynamicsBackend could define some kind of placeholder gate between coupled qubits or otherwise enforce the coupling map? Perhaps this is overkill if the only goal is to avoid an error with a transpiler pass that expects a coupling map and there are no other reasons to need an accurate coupling map for the DynamicsBackend.

Reply:
Thanks @wshanks for investigation. Yes, your summary looks correct. Target generates the coupling map on the fly based on stored instructions (one provided by the backend is just ignored). I think this behavior is weird, because the coupling edge doesn't appear unless user adds calibration even though two nodes are physically connected.

Probably you could do small hack for `DynamicsBackend`. Namely, you can override [coupling_map](https://github.com/Qiskit/qiskit-terra/blob/fa7ac688fdce3ecc27fb7e13d8b2c3f8553b3ff8/qiskit/providers/backend.py#L387-L390) not to call `Target.build_coupling_map` method. You can generate the coupling map based on device Hamiltonian (usually parsing Hamiltonian is really tough). Alternatively you can simply assume full connectivity and returns `CouplingMap(itertools.product(range(n_qubit), range(n_qubit)))`.

Reply:
Okay - I've just added a `coupling_map` option to the backend, and implemented the `coupling_map` property, in the following way:
- If the option is set to `"complete"`, calling `backend.coupling_map` will return a `CouplingMap` representing the complete graph. I had to modify your suggested code to avoid having edges of the form `(0, 0)`.
- If the option is set to `"target"`, `backend.coupling_map` will call `target.build_coupling_map()` as before.
- If the option is set to a `CouplingMap` instance, `backend.coupling_map` will just return that.
I've also added error raising and tests to check for the behaviour of this option. The default is to return the complete graph.

Unfortunately, after implementing this, and removing the `CXGate` hack, I still got the error that the `cr_gate` cannot be flipped. So, this doesn't appear to actually solve the issue. In the section for the `CrossResonanceHamiltonian`, I've re-added a line adding the `CRPulseGate` to the `target`. 

Any other ideas @wshanks @nkanazawa1989 ? If not I'm fine with leaving things the way they are currently.

Reply:
Hmm, you overrode the Backend's coupling map, but I would guess that for the transpiler issues we have been hitting here it is the Target's coupling map gets used.

Reply:
The default behaviour is for the coupling map to be complete - I thought this would resolve the problem.

Ah but I guess you're saying that whatever is looking up the coupling map is directly accessing the coupling map `Target`?

I think I might just revert these changes and add the `CRPulseGate` to the instructions as before. It would be nice to not have to do this, but this seems like a much cleaner hack than messing around too much with the `DynamicsBackend` class itself.

Reply:
Alright I've reverted this - unless there are serious objections to this I think the hack of adding this gate is the preferred solution.

Reply:
I think using a hack here is the best option that I know of. Ideally, qiskit-experiments will allow more flexibility with transpilation in the future which could avoid the issue. The one thing I still wonder about is if there is set of transpiler options that could disable whatever is causing the error (since we know the circuit should run fine on DynamicsBackend without any transpilling, we just need the `transpile` call that qiskit-experiments does not to change anything).

Also, I think there might be a transpiler bug here. I will look into it a little more and report something to terra if it makes sense.

Reply:
@wshanks @nkanazawa1989 - if this works for you, please approve. It is the last PR before for the next release :).

Reply:
Terra issue about the transpiler problem we were seeing: https://github.com/Qiskit/qiskit-terra/issues/9783